src/userobjects/MDRunBase.C

```diff
@@ -39,35 +39,78 @@ validParams<MDRunBase>()
   InputParameters params = validParams<GeneralUserObject>();
   params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_BEGIN;
   params.suppressParameter<ExecFlagEnum>("execute_on");
-
+  params.addParam<MultiMooseEnum>("md_particle_properties",
+                                  MDRunBase::mdParticleProperties(),
+                                  "Properties of MD particles to be obtained and stored.");
+  params.addRangeCheckedParam<Real>(
+      "max_granular_radius",
+      0,
+      "max_granular_radius>=0",
+      "Maximum radius of granular particles. This can be important for partitioning.");
   params.addClassDescription(
       "Base class for execution of coupled molecular dynamics MOOSE calculations.");
   return params;
 }
 
 MDRunBase::MDRunBase(const InputParameters & parameters)
   : GeneralUserObject(parameters),
+    _properties(getParam<MultiMooseEnum>("md_particle_properties")),
+    _granular(_properties.contains("radius")),
     _mesh(_subproblem.mesh()),
     _dim(_mesh.dimension()),
     _nproc(_app.n_processors()),
     _bbox(_nproc)
 {
+  // if the calculation is granular, max_particle_radius
+  if (_granular)
+  {
+    if (!parameters.isParamSetByUser("max_granular_radius"))
+      paramError("max_granular_radius",
+                 "max_granular_radius must be set for granular calculations");
+    _max_granular_radius = getParam<Real>("max_granular_radius");
+  }
+
+  // check _properties array and MooseEnum for consistency
+  if (N_MD_PROPERTIES != mdParticleProperties().getNames().size())
+    mooseError("Mismatch of MD particle property enum and property array. Report on github.");
```

Comment:
Oh, does that mean a very specific dump format is required?

Reply:
Yes, must be `xyz` format.